File: AmpedBiz/AmpedBiz.Service/Dto/Measure.cs
namespace AmpedBiz.Service.Dto
{
    public class Measure
    {
        public decimal Value { get; set; }

        public UnitOfMeasure Unit { get; set; }

        public Measure() { }

        public Measure(decimal value, UnitOfMeasure unit)
        {
            this.Value = value;
            this.Unit = unit;
        }
    }
}

File: AmpedBiz/AmpedBiz.Web/src/common/models/measure.ts
import { UnitOfMeasure } from "./unit-of-measure";

export type Measure = {
  value?: number;
  unit?: UnitOfMeasure;
}

export const getValue = (measure: Measure) => measure && measure.value || 0;

export const getUnitId = (measure: Measure) => measure && measure.unit && measure.unit.id || '';

